### PR TITLE
Add events on switching landscape/imagelight box

### PR DESCRIPTION
**Pull request by sunriseshoapeng**

### CODE REVIEW

File: VideoPlayer.js [hunk 1]
```diff
@@ -665,6 +670,15 @@ export default class VideoPlayer extends Component {
     this.setState(state);
   }
 
+  seekAndPlay(time){
+    let state = this.state;
+    state.currentTime = time;
+    this.player.ref.seek(time);
+    this.setState(state);
+  }
```
Comment (MattHorriganSunrise) [inline VideoPlayer.js]: ? theres already a seekTo method.

File: VideoPlayer.js [hunk 2]
```diff
@@ -1302,8 +1317,8 @@ export default class VideoPlayer extends Component {
             />
             {this.renderError()}
             {this.renderLoader()}
-            {this.renderTopControls()}
-            {this.renderBottomControls()}
+            {!this.props.hideAllControls ? this.renderTopControls() : null}
+            {!this.props.hideAllControls ? this.renderBottomControls() : null}
```
Comment (MattHorriganSunrise) [inline VideoPlayer.js]:
```js
{!this.props.hideAllControls && (
  <>
   {this.renderTopControls()}
   {this.renderBottomControls()}
  <>
)}

Comment (MattHorriganSunrise) [reply VideoPlayer.js]: or: 
```js
{!this.props.hideAllControls && this.renderTopControls()}
{!this.props.hideAllControls && this.renderBottomControls()}

Comment (sunriseshoapeng) [reply VideoPlayer.js]: I gotta get used the the && operator aha. Im just so used to using ? : . It's just more readable for me aha

Comment (MattHorriganSunrise) [reply VideoPlayer.js]: ? : is more readable for you than &&, thats weird lol

Comment (sunriseshoapeng) [reply VideoPlayer.js]: Ahaha yeah its all good, Ill start doing && operator

File: VideoPlayer.js [hunk 3]
```diff
@@ -276,6 +282,7 @@ export default class VideoPlayer extends Component {
    * new page.
    */
   _onEnd() {
+    this.setState({})
```
Comment (MattHorriganSunrise) [inline VideoPlayer.js]: You have conflicts, and you can probably remove this.

File: VideoPlayer.js [hunk 4]
```diff
@@ -279,6 +285,7 @@ export default class VideoPlayer extends Component {
    * new page.
    */
   _onEnd() {
+    this.setState({isEnded: true});
```
Comment (MattHorriganSunrise) [inline VideoPlayer.js]: this isn't needed anymore, why add this?

File: VideoPlayer.js [hunk 5]
```diff
@@ -213,6 +214,11 @@ export default class VideoPlayer extends Component {
     // This triggers channel Avatar Channel & Follow Button [Landscape View]
     this.props.streamLandscapeStore.isShadowOverlayOn = true;
 
+    // Seeking to previous location
+    if(this.props.streamStore.seekVideo){
+      this.seekTo(this.props.streamStore.seekToTime)
+    }
```
Comment (MattHorriganSunrise) [inline VideoPlayer.js]: I know your prettier doesn't work on this but can we make a habit of doing this for spaces:
```js
if () {

File: VideoPlayer.js [hunk 6]
```diff
@@ -233,7 +239,12 @@ export default class VideoPlayer extends Component {
     let state = this.state;
     if (!state.scrubbing) {
       state.currentTime = data.currentTime;
-      this.props.streamStore.currentVideoTime = state.currentTime;
+
+      if(!this.props.isLive){
+        if(state.currentTime !== 0){
```
Comment (MattHorriganSunrise) [inline VideoPlayer.js]: 0 coercion to false.
```js
if (state.currentTime) {

**Review by MattHorriganSunrise — APPROVED** [review]

File: VideoPlayer.js (hunk 6 above)
Comment (sunriseshoapeng) [reply VideoPlayer.js]: Nice, ill keep that in mind

**Pull request merged**